Sources/RInAppMessaging/InAppMessagingModule.swift

```diff
@@ -93,6 +93,9 @@ internal class InAppMessagingModule: ErrorDelegate, CampaignDispatcherDelegate,
         accountRepository.setPreference(preference)
 
         guard isInitialized else {
+            if accountRepository.updateUserInfo() {
+                campaignRepository.loadCachedData()
```

Comment:
So is the `registerPreference` API not getting retained and called at a later time after initialization, as per this PR #302 ?

Because the logic to load the cached data is supposed to be done by the `checkUserChanges` (line 102) after SDK is initialized.

Reply:
Yes , [#302](https://github.com/rakutentech/ios-inappmessaging/pull/302) retains the log events and not the registerPreference . So in case of RMC, if the registerPreference() and logEvents() are called before SDK is initialized, the cached user data is not loaded .

Reply:
In the #302, both `registerPreference` & `logEvents` has been retained. The issue was local cached campaign was not loaded in campaign repository. 

Line-102 `checkUserChanges` will be call when the configuration is completed. `checkUserChanges` also calls the `campaignRepository.loadCachedData()` and ping request. No need to call ping request in this case as configuration is not finished.

Reply:
Updated RCA and fix in the PR description.

Reply:
Thanks, but it looks like it should be loaded in `CampaignRepository.init`: https://github.com/rakutentech/ios-inappmessaging/blob/master/Sources/RInAppMessaging/Repositories/CampaignRepository.swift#L78. I am not sure why it didn't work. Anyway, just please test it thoroughly.

Reply:
Yes, you are correct. User details is not available when initialize the `CampaignRepository` so we are updating the local cached data once the user details is updated by calling `registerPreference`.

Reply:
In demo app we are calling `registerPreference` method before we `configure`. 
The Readme here says `Ensure that `registerPreference()` is called after `configure()` and before any `logEvent()` method`.

Reply:
Yes, this was already discussed. @SoumenRautray will update the instruction. This changes are introduced to support RMC SDK integration.